listenbrainz_spark/request_consumer/request_consumer.py

```diff
@@ -54,26 +58,48 @@ def get_response(self, request):
             return None
 
 
-    def push_to_response_queue(self, response):
+    def push_to_result_queue(self, result):
         while True:
             try:
-                self.response_channel.basic_publish(
+                self.result_channel.basic_publish(
                     exchange=current_app.config['SPARK_RESULT_EXCHANGE'],
                     routing_key='',
-                    body=ujson.dumps(response),
+                    body=json.dumps(result),
                     properties=pika.BasicProperties(delivery_mode = 2,),
                 )
                 break
             except pika.exceptions.ConnectionClosed:
+                self.connect_to_rabbitmq()
                 time.sleep(1)
+            except pika.exceptions.ChannelClosed:
+                self.result_channel = self.rabbitmq.channel()
+                self.result_channel.exchange_declare(exchange=current_app.config['SPARK_RESULT_EXCHANGE'], exchange_type='fanout')
+
 
 
     def callback(self, channel, method, properties, body):
+        current_app.logger.info("Processing new request...")
         request = json.loads(body.decode('utf-8'))
-        response = self.get_response(request)
-        if response:
-            self.push_to_response_queue(response)
-        channel.basic_ack(delivery_tag=method.delivery_tag)
+        current_app.logger.info("Calculating result...")
+        result = self.get_result(request)
+        current_app.logger.info("Done!")
+        if result:
+            current_app.logger.info("Pushing to result queue...")
+            self.push_to_result_queue(result)
+            current_app.logger.info("Done!")
+        while True:
+            try:
+                self.request_channel.basic_ack(delivery_tag=method.delivery_tag)
+                break
+            except pika.exceptions.ChannelClosed:
+                self.request_channel = self.rabbitmq.channel()
+                self.request_channel.exchange_declare(exchange=current_app.config['SPARK_REQUEST_EXCHANGE'], exchange_type='fanout')
```

Comment:
What is going on here with the indent?

Reply:
what indent problem?

Reply:
oh, I see -- if the window is too small it looks like the indent is off.